.github/workflows/scheduled.yml

```diff
@@ -0,0 +1,112 @@
+name: Scheduled regeneration
+
+on:
+  schedule:
+    # Once every hour
+    - cron:  '*/60 * * * *'
+
+jobs:
+  generate-macos:
+    if: startsWith( github.repository, 'Homebrew/' )
+    runs-on: macos-latest
+    env:
+      HOMEBREW_COLOR: 1
+      HOMEBREW_NO_ANALYTICS: 1
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Homebrew
+        run: |
+          brew update-reset $(brew --repository)
+
+          git config --global user.name "BrewTestBot"
+          git config --global user.email "homebrew-test-bot@lists.sfconservancy.org"
+
+          # set up analytics
+          echo ${{ secrets.ANALYTICS_JSON_KEY }} > ~/.homebrew_analytics.json
+
+      - name: Update data for homebrew/core
+        run: |
+          brew tap homebrew/core
+          brew update-reset $(brew --repository homebrew/core)
+
+          # run Rakefile
+          /usr/bin/rake
+
+          # commit and push generated files
+          git add _data/formula api/formula formula
+
+          if ! git diff --no-patch --exit-code HEAD -- _data/analytics _data/formula api/formula formula cask; then
+            git commit -m "formula: updates from homebrew/core" _data/analytics _data/formula api/formula formula
+            git fetch
+            git rebase origin/master
+            git push
+          fi
+
+      - name: Update data for homebrew/cask
+        run: |
+          brew tap homebrew/cask
+          brew update-reset $(brew --repository homebrew/cask)
+
+          # run Rakefile
+          /usr/bin/rake cask
+
+          # commit and push generated files
+          git add _data/cask api/cask cask
+
+          if ! git diff --no-patch --exit-code HEAD -- _data/cask api/cask cask; then
+            git commit -m "cask: updates from homebrew/cask" _data/cask api/cask cask
+            git fetch
+            git rebase origin/master
+            git push
+          fi
+
+  generate-linux:
```

Comment:
Thoughts on making this a matrix job so it can share more code?

Reply:
(check the homebrew-core or homebrew-test-bot repos for prior art for e.g. the setup)

Reply:
Any ideas for a container to apply to only one OS in a matrix? [No luck](https://github.com/EricFromCanada/formulae.brew.sh/actions/runs/94490949) with that so far…

Reply:
No ideas sorry 😭